WebService.Test/v1/Controllers/DeviceModelPropertiesControllerTest.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controllers;
+using Moq;
+using WebService.Test.helpers;
+using Xunit;
+
+namespace WebService.Test.v1.Controllers
+{
+    public class DeviceModelPropertiesControllerTest
+    {
+        private readonly Mock<IDeviceModels> deviceModelsService;
+        private readonly DeviceModelPropertiesController target;
+
+        public DeviceModelPropertiesControllerTest()
+        {
+            this.deviceModelsService = new Mock<IDeviceModels>();
+            this.target = new DeviceModelPropertiesController(this.deviceModelsService.Object);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ItReturnsTheListOfPropertyNames()
+        {
+            // Arrange
+            var properties = new List<string>
+            {
+                "Type", "Firmware" , "Location" ,"Model", "Latitude" ,"Longitude"
+            };
+
+            this.deviceModelsService
+                .Setup(x => x.GetPropertyNamesAsync())
+                .ReturnsAsync(properties);
+
+            // Act
+            var result = this.target.GetAsync().Result;
+
+            // Assert
+            Assert.Equal(properties.Count, result.Items.Count);
+            foreach (var resultItem in result.Items)
+            {
+                Assert.Contains(resultItem, properties);
+            }
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void ItThrowsExceptionWhenGetPropertyNamesFailed()
+        {
+            // Arrange
+            this.deviceModelsService
+                .Setup(x => x.GetPropertyNamesAsync())
+                .ThrowsAsync(new Exception());
+
+            // Act & Assert
+            Assert.ThrowsAsync<Exception>(
```

Comment:
This doesn't verify that the exception thrown is the one configured on line 55, and the test could pass for the wrong reason, hiding a bug.
That's why we use a helper class `SomeException` which doesn't exist in the application.  Configuring `GetPropertyNameAsync` to throw `SomeException` allows us to verify here that the exception thrown is exactly `SomeException` and not something else.
It's probably worth adding this explanation to `SomeException` for future reference